Clarify orphan sweep behavior to agentic memory retention doc (#12895)
opensearch-project/documentation-website: _ml-commons-plugin/agentic-memory-retention.md

- A memory that has passed its age or count limit is still returned in search results until the job next runs and deletes it.
- The `retention_days` limit is measured from a memory's `last_updated_time`. Adding messages to a session advances this timestamp, so active conversations are retained. Pinning a memory does not advance the timestamp.
- When a memory type sets both `retention_days` and `max_count`, a memory is deleted if it exceeds either limit. The count limit applies regardless of age: if a container holds more than `max_count` non-pinned memories, the oldest are deleted until `max_count` remain, even if they are newer than `retention_days`.

> - Each run deletes at most 50,000 memories per memory type, per container — applied independently to sessions, long-term, and history (working memory is not subject to this cap). A larger backlog is reduced over successive runs, so a container that far exceeds its `max_count` may take several runs to reach the limit.

Reviewer: ```suggestion
- Each run deletes at most 50,000 memories for each memory type (`sessions`, `long-term`, and `history`) in a container. A larger backlog is reduced over successive runs, so a container that far exceeds its `max_count` may take several runs to reach the limit.
```
